src/main/java/teammates/ui/controller/InstructorCoursesPageData.java

```diff
@@ -1,35 +1,149 @@
+                archivedCourses.getRows().add(row);
+            }
+        }
+        
+        return archivedCourses;
+    }
+    
+    private ActiveCoursesTable extractActiveCourses(List<CourseDetailsBundle> courseBundles) {
+        ActiveCoursesTable activeCourses = new ActiveCoursesTable();
+        Logic logic = new Logic();
+        
+        CourseDetailsBundle.sortDetailedCoursesByCourseId(courseBundles);
+        
+        int idx = -1;
+        for (CourseDetailsBundle courseBundle : courseBundles) {
+            CourseAttributes course = courseBundle.course;
+
+            InstructorAttributes curInstructor = logic.getInstructorForGoogleId(course.id, account.googleId);
+
+            if (!Logic.isCourseArchived(course.id, curInstructor.googleId)) {
+                idx++;
+                
+                String actionsParam = "<a class=\"btn btn-default btn-xs t_course_enroll" + idx + "\""
+                                                + "href=\"" + getInstructorCourseEnrollLink(courseBundle.course.id) + "\""
+                                                + "data-toggle=\"tooltip\" data-placement=\"top\" title=\"" + Const.Tooltips.COURSE_ENROLL + "\""
+                                                + (curInstructor.isAllowedForPrivilege(Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_STUDENT) ? ""
+                                                                                                                                               : "disabled=\"disabled\"")
+                                      + ">Enroll</a>";
+                            
+                actionsParam += "<a class=\"btn btn-default btn-xs t_course_view" + idx + "\""
+                                    + "href=\"" + getInstructorCourseDetailsLink(courseBundle.course.id) + "\""
+                                    + "data-toggle=\"tooltip\" data-placement=\"top\" title=\"" + Const.Tooltips.COURSE_DETAILS + "\""
+                                + ">View</a>";
+                        
+                actionsParam += "<a class=\"btn btn-default btn-xs t_course_edit" + idx + "\""
+                                    + "href=\"" + getInstructorCourseEditLink(courseBundle.course.id) + "\""
+                                    + "data-toggle=\"tooltip\" data-placement=\"top\" title=\"" + Const.Tooltips.COURSE_EDIT + "\""
+                                + ">Edit</a>";
+
+                actionsParam += "<a class=\"btn btn-default btn-xs t_course_archive" + idx + "\""
+                                    + "href=\"" + getInstructorCourseArchiveLink(courseBundle.course.id, true, false) + "\""
+                                    + "data-toggle=\"tooltip\" data-placement=\"top\" title=\"" + Const.Tooltips.COURSE_ARCHIVE + "\""
+                                + ">Archive</a>";
+
+                actionsParam += "<a class=\"btn btn-default btn-xs t_course_delete" + idx + "\""
```

Comment:
Try to keep the usage of HTML in Java to a minimum.
I believe we're hoping to remove them entirely eventually, so let's not introduce more.